File: proyecto/src/main/java/com/example/demo/model/entidad/Pedido.java
package com.example.demo.model.entidad;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "Pedido")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Pedido {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID_Pedido")
    private Long idPedido;

    @ManyToOne
    @JoinColumn(name = "ID_Usuario", nullable = false)
    private Usuario usuario;

    @ManyToOne
    @JoinColumn(name = "ID_InfoEnvio", nullable = false)
    private InformacionEnvio informacionEnvio;

    @NotBlank(message = "El número de pedido es obligatorio")
    @Column(name = "numero_pedido", nullable = false, unique = true, length = 20)
    private String numeroPedido;

    @Column(name = "fecha_pedido")
    private LocalDateTime fechaPedido = LocalDateTime.now();

    @Column(name = "subtotal", nullable = false)
    private Double subtotal = 0.00;

    @Column(name = "igv", nullable = false)
    private Double igv = 0.00;

    @Column(name = "total", nullable = false)
    private Double total = 0.00;

    @Enumerated(EnumType.STRING)
    @Column(name = "estado", nullable = false)
    private Estado estado = Estado.Pendiente;

    public enum Estado {
        Pendiente, Pagado, Entregado, Cancelado
    }

    @Enumerated(EnumType.STRING)
    @Column(name = "metodo_pago", nullable = false)
    private MetodoPago metodoPago = MetodoPago.Efectivo;

    public enum MetodoPago {
        Efectivo, Tarjeta, Transferencia, Yape, Plin
    }

    @Column(name = "observaciones", columnDefinition = "TEXT")
    private String observaciones;

    @Column(name = "usuario_registro")
    private Integer usuarioRegistro;
}

File: Ticket/src/app/model/pedido.ts
export interface Pedido {
  idPedido: number;
  idUsuario: number;
  numeroPedido: string;
  fechaRegistro: string;
  estado: 'Pendiente' | 'Pagado' | 'Entregado' | 'Cancelado';
  metodoPago: string;
  total: number;
  envio: {
    idInfoEnvio: number;
    direccion: string;
    distrito: string;
    ciudad: string;
    nombreDestinatario: string;
  };
  detalles: PedidoDetalle[];
}

export interface PedidoDetalle {
  idDetalle: number;
  idProducto: number;
  nombreProducto: string;
  cantidad: number;
  precioUnitario: number;
  subtotal: number;
}

export interface PedidoResponse {
  idPedido: number;
  numeroPedido: string;
  fechaRegistro: string;
  estado: string;
  metodoPago: string;
  total: number;
  idInfoEnvio: number;
  direccion: string;
  distrito: string;
  ciudad: string;
  nombreDestinatario: string;
  detalles: {
    idDetalle: number;
    idProducto: number;
    nombreProducto: string;
    cantidad: number;
    precioUnitario: number;
    subtotal: number;
  }[];
}